File: projectThree/cpp/State.cpp
#include "state.h"

State::State(int q, int d, int n, bool c) {
    this->quarters = q;
    this->dimes = d;
    this->nickels = n;
    this->change = c;
    this->value = 0;
}

State::State(State *s, int q, int d, int n, int v, bool c) {
    this->quarters = s->getQuanCoins('q') + q;
    this->dimes = s->getQuanCoins('d') + d;
    this->dimes = s->getQuanCoins('n') + n;
    this->change = c;
    this->value = s->getValue() + this->makeValue(q, d, n) - v;
}

int State::makeValue(int q, int d, int n) {
    return ((q * 25) + (d * 10) + (n * 5));
}

int State::getQuanCoins(char c) {
    int qCoins = 0;
    switch(c) {
        case 'q':
            qCoins = this->quarters;
            break;
        case 'd':
            qCoins = this->dimes;
            break;
        case 'n':
            qCoins = this->nickels;
            break;
    }
    return qCoins;
}

int State::getValue() {
    return this->value;
}

bool State::isPressed() {
    return this->change;
}

File: projectTwo/State.java
import java.text.DecimalFormat;

public class State {
    DecimalFormat df = new DecimalFormat("#.##");
    int quarters, dimes, nickels, value;
    boolean change;
    public State(int q, int d, int n, boolean c) {
        this.quarters = q;
        this.dimes = d;
        this.nickels = n;
        this.change = c;
        this.value = 0;
    }

    public State(State s, int q, int d, int n, int v, boolean c) {
        this.quarters = s.getQuanCoins('q') + q;
        this.dimes = s.getQuanCoins('d') + d;
        this.nickels = s.getQuanCoins('n') + n;
        this.change = c;
        this.value = s.getValue() + this.makeValue(q, d, n) - v;
    }

    public int makeValue(int q, int d, int n) {
        //System.out.println(q + ", " + d + ", " + n);
        return ((q * 25) + (d * 10) + (n * 5)); 
    }

    public int getQuanCoins(char i) {
        int quant = 0;
        switch(i) {
            case 'q':
                quant = this.quarters;
                break;
            case 'd':
                quant = this.dimes;
                break;
            case 'n':
                quant = this.nickels;
                break;
        }
        return quant;
    }

    public int getValue() {
        return this.value;
    }

    public boolean isPressed() {
        return this.change;
    }

    @Override
    public String toString() {
        return "Quarters: " + this.quarters + " Dimes: " + this.dimes + " Nickels: " + this.nickels + " Value: $" + df.format((float)this.value/100);
    }
}